support-lib/wasm/djinni_wasm.cpp

```diff
@@ -60,7 +60,13 @@ const em::val& JsProxyBase::_jsRef() const {
 
 void JsProxyBase::checkError(const em::val& v) {
     if (v.instanceof(em::val::global("Error"))) {
-        throw JsException(v);
+        auto cppExceptionPtr = v["cppExceptionPtr"];
+        if (!cppExceptionPtr.isUndefined()) {
+            std::exception_ptr* exptr = reinterpret_cast<std::exception_ptr*>(cppExceptionPtr.as<int>());
+            std::rethrow_exception(*exptr);
+        } else {
+            throw JsException(v);
+        }
```

Comment:
Yes! This is exactly why I was prompting to add the tests for the extra cases, it didn't seem like we had handled this case